src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -1914,3 +1914,65 @@ void CLRCriticalSection::Leave()
 {
     pthread_mutex_unlock(&m_cs.mutex);
 }
+
+#if defined(_X86_) || defined(_AMD64_)
+REDHAWK_PALEXPORT uint32_t REDHAWK_PALAPI getcpuid(uint32_t arg, unsigned char result[16])
+{
+    DWORD eax;
+#if defined(_X86_)
+    __asm("  xor %%ecx, %%ecx\n" \
+          "  cpuid\n" \
+          "  mov %%eax, 0(%[result])\n" \
```

Comment:
Nit: You can let the C/C++ compiler to generate these movs for you by telling it that the result is in registers. Something like:

`"=a" ((uint32_t&)result[0]), "=b" ((uint32_t&)result[4]), "=c" ((uint32_t&)result[8]), "=d" ((uint32_t&)result[12])`

Once you do that, you should be able to omit them from the clobber section.

Reply:
FWIW, this is what cpuid looks like in clang: https://github.com/llvm-mirror/clang/blob/master/lib/Headers/cpuid.h#L223

Reply:
I ended up just stealing all of this from CoreCLR because I can't really test this on X86 with CoreRT and it's known to work there on CoreCLR.